google-cloud-clients/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java

```diff
@@ -285,10 +290,24 @@ public TableResult getQueryResults(QueryResultsOption... options)
     QueryResponse response =
         waitForQueryResults(
             DEFAULT_JOB_WAIT_SETTINGS, waitOptions.toArray(new QueryResultsOption[0]));
-    if (response.getSchema() == null) {
-      throw new JobException(getJobId(), response.getErrors());
+
+    // Get the job resource to determine if it has errored.
+    Job job = this;
+    if (job.getStatus() == null || job.getStatus().getState() != JobStatus.State.DONE) {
+      job = reload();
+    }
+    if (job.getStatus() != null && job.getStatus().getError() != null) {
+      throw new JobException(
+          getJobId(), ImmutableList.copyOf(job.getStatus().getExecutionErrors()));
```

Comment:
I don't completely understand why we need to reload the job. Can we infer what we need from the response?

The reason I'm asking is that some folks has complained about high latency in the past.

Reply:
No, as far as I can tell there is no way to determine that a job has failed from `getQueryResults()`. Previously this code looked for a missing schema, but that can happen for certain DDL and related queries now.